src/HazelcastClient.ts

```diff
@@ -410,28 +487,20 @@ export class HazelcastClient {
         return this.proxyManager.removeDistributedObjectListener(listenerId);
     }
 
-    /** @internal */
-    getLockReferenceIdGenerator(): LockReferenceIdGenerator {
-        return this.lockReferenceIdGenerator;
-    }
-
     /** @internal */
     getErrorFactory(): ClientErrorFactory {
         return this.errorFactory;
     }
 
-    /** @internal */
-    getLoadBalancer(): LoadBalancer {
-        return this.loadBalancer;
-    }
-
-    /** @internal */
-    getClusterFailoverService(): ClusterFailoverService {
-        return this.clusterFailoverService;
-    }
+    /**
+     * Shuts down this client instance.
+     */
+    shutdown(): Promise<void> {
+        if (!this.lifecycleService.isRunning()) {
+            return Promise.resolve();
+        }
+        this.lifecycleService.onShutdownStart();
 
-    /** @internal */
-    doShutdown(): Promise<void> {
```

Comment:
Do we want to keep doShutdown()? merged it with shutdown()

Reply:
I don't think we need to keep it. It's whole purpose was to act as a private callback to be invoked from `LifecycleService`.